### DISCUSSION TITLE
Failed to access the Knowledge POST API using JSON object

### DISCUSSION THREAD

**Discussion by optimono** [langgenius/dify#10581] [Help]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### Content

using curl is OK:
`
curl --location --request POST 'http://{server_ip}/v1/datasets/{dataset_id}/document/create_by_file' --header 'Authorization: Bearer {api_key}' --form 'data={"name":"assistant","indexing_technique":"high_quality","process_rule":{"rules":{"pre_processing_rules":[{"id":"remove_extra_spaces","enabled":true},{"id":"remove_urls_emails","enabled":true}],"segmentation":{"separator":"###","max_tokens":500}},"mode":"custom"}};type=text/plain' --form 'file=@"{file_path}"'
`

Referred issue answer [https://github.com/langgenius/dify/issues/2589#issuecomment-1968046442](url), use the following python script:
```
url = f'http://{server_ip}/v1/datasets/{dataset_id}/document/create_by_file'
headers = {
  'Authorization': f'Bearer {api_key}',
}
data = {
  "name": "assistant",
  "indexing_technique": "high_quality",
  "process_rule": {
	  "rules": {
		  "pre_processing_rules": [
			  {
				  "id": "remove_extra_spaces",
				  "enabled": True
			  },
			  {
				  "id": "remove_urls_emails",
				  "enabled": True
			  }
		  ],
		  "segmentation": {
			  "separator": "###",
			  "max_tokens": 500
		  }
	  },
	  "mode": "custom"
  }
}

files = {
  'data': ('', json.dumps(data), 'text/plain'),
  'file': (file_name, open(file_path, 'rb'))
}
response = requests.post(url, headers=headers, files=files)
```

Returned error:
`400
{"code": "too_many_files", "message": "Only one file is allowed.", "status": 400}`

How to resolve this?

**Comment by optimono** [comment]
Any help? T_T

**Comment by FujieBao** [comment]
Same problem! using curl works but when I use javascript sent the POST request meeting the same problem.
{code: "too_many_files", message: "Only one file is allowed.", status: 400}
